pyfstat/make_sfts.py

```diff
@@ -943,6 +941,175 @@ def __init__(
         )
 
 
+class TransientLineWriter(Writer):
+    """Inject a transient line into some data.
+
+    A transient line is defined as a constant amplitude and frequency excess power lasting
+    for a (not necesarilly) finite time span of an observing run.
+
+    In practice, it corresponds to a CW without Doppler or antenna-patern-induced amplitude modulation.
+
+    This uses MFD_v4
+    """
+
+    required_mfd_line_parameters = [
+        "Freq",
+        "phi0",
+        "h0",
+        "cosi",
+        "transientWindowType",
+        "transientStartTime",
+        "transientTau",
+    ]
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        if self.detectors is None:
+            raise ValueError("MakeFakeData_v4 requires detector name to be given")
+        elif len(self.detectors.split(",")) > 1:
+            raise NotImplementedError(
+                "MakeFakeData_v4 does not support more than one detector at a time. "
+                "Please, do it properly"
+            )
+
+    def correct_line_amplitude(self, use_effective_h0, scale_by_duration):
+        """
+        Modify the amplitude parameter h0 to include effects from polarization (cosi)
+        or account for shorter signals.
+
+        use_effective_h0: bool
+            If true, use h0_eff = h0 * sqrt(cosi^4 + 6 * cosi^2 + 1)
+        scale_by_duration: bool
+            If true, scale h0 up by sqrt(t_obs/tau)
+        """
+
+        if use_effective_h0:
+            eta = self.signal_parameters["cosi"]
+            eta2 = eta * eta
+            self.signal_parameters["h0"] *= np.sqrt(eta2 * eta2 + 6 * eta2 + 1)
+        if scale_by_duration:
+            self.signal_parameters["h0"] *= np.sqrt(self.duration / self.transientTau)
+
+    def _parse_args_consistent_with_mfd(self):
+        """
+        Adapt input arguments.
+        Take care of minor inconsistencies between MFD_v4 and MFD_v5
+        """
+        super()._parse_args_consistent_with_mfd()
+
+        if any(
+            key not in self.required_mfd_line_parameters
+            for key in self.signal_parameters
+        ):
+            logging.warning(
+                "Transient lines only require the following parameters:\n"
+                f"{self.required_mfd_line_parameters}.\n"
+                "Any other parameter will be purged from this class now"
+            )
+            params_to_purge = list(
+                set(self.signal_parameters) - set(self.required_mfd_line_parameters)
+            )
+            print("Purging: {}".format(params_to_purge))
+            for key in params_to_purge:
+                self.signal_parameters.pop(key)
+
+        if "transientTau" in self.signal_parameters:
+            self.signal_parameters["transientTauDays"] = (
+                self.signal_parameters.pop("transientTau") / 86400.0
+            )
```

Comment:
Probably ok if tests pass, but shouldn't this be an int?

Reply:
This is one of those places where v4 and v5 differ.

MFDv4 parses parameters using `XLALRegisterUvarMember`, asking for `float` (well, `double`)
```
  XLALRegisterUvarMember(transientWindowType,  STRING, 0, OPTIONAL, "Type of transient signal window to use. ('none', 'rect', 'exp').");
  XLALRegisterUvarMember(  transientStartTime,   REAL8, 0, OPTIONAL, "GPS start-time 't0' of transient signal window.");
  XLALRegisterUvarMember(  transientTauDays,     REAL8, 0, OPTIONAL, "Timescale 'tau' of transient signal window in days.");
   ```
 while we use MFDv5 in such a way that the parsing is done through a `cff` file, which ends up being read in `XLALReadPulsarParams`
 ```
    // ---------- transientWindow_t ----------
 
   // ----- type
   char *transientWindowType = NULL; BOOLEAN have_transientWindowType;
   XLAL_CHECK ( XLALReadConfigSTRINGVariable ( &transientWindowType, cfgdata, secName, "transientWindowType", &have_transientWindowType ) == XLAL_SUCCESS, XLAL_EFUNC );
   // ----- t0
   UINT4 transientStartTime = 0; BOOLEAN have_transientStartTime;
   XLAL_CHECK ( XLALReadConfigUINT4Variable ( &transientStartTime, cfgdata, secName, "transientStartTime", &have_transientStartTime ) == XLAL_SUCCESS, XLAL_EFUNC );
   // ----- tau (still keeping deprecated Days variant for backwards compatibility, for now)
   UINT4 transientTau = 0; BOOLEAN have_transientTau;
   XLAL_CHECK ( XLALReadConfigUINT4Variable ( &transientTau, cfgdata, secName, "transientTau", &have_transientTau ) == XLAL_SUCCESS, XLAL_EFUNC );
   REAL8 transientTauDays = 0; BOOLEAN have_transientTauDays;
   XLAL_CHECK ( XLALReadConfigREAL8Variable ( &transientTauDays, cfgdata, secName, "transientTauDays", &have_transientTauDays ) == XLAL_SUCCESS, XLAL_EFUNC );
 ```
where `transientTau` is `UINT4`. 

Funnily enough, `transientTauDays` is still supported, for now :wink:.